From: John W. Linville <linville@redhat.com>
Date: Thu, 4 Sep 2008 15:32:22 -0400
Subject: [wireless] iwlwifi: fix busted tkip encryption
Message-id: 20080904193222.GL20038@redhat.com
O-Subject: [RHEL5 PATCH] iwlwifi: fix busted tkip encryption after iwl5000 support update
Bugzilla: 438388

On Thu, Aug 28, 2008 at 01:59:00PM -0400, John W. Linville wrote:
> Yet another monster patch...this one updates the iwlwifi drivers to
> match what was released in 2.6.27-rc3 plus some even later patches
> (the ones from the current upstream process discussion on-going in
> linux-wireless for anyone watching that).  This also renames the
> iwl4965 driver to iwlagn (MODULE_ALIAS added) and adds support for
> the iwl5x00 adapters.
>
> BZ438388

This patch inadvertently broke TKIP encryption for this driver.
The patch below restores correct functionality.

Sorry for the mess!

John

diff --git a/drivers/net/wireless/iwlwifi/iwl-sta.c b/drivers/net/wireless/iwlwifi/iwl-sta.c
index b106436..03447d6 100644
--- a/drivers/net/wireless/iwlwifi/iwl-sta.c
+++ b/drivers/net/wireless/iwlwifi/iwl-sta.c
@@ -207,6 +207,7 @@ static int iwl_set_ccmp_dynamic_key_info(struct iwl_priv *priv,
 
 	spin_lock_irqsave(&priv->sta_lock, flags);
 	priv->stations[sta_id].keyinfo.alg = keyconf->alg;
+	priv->stations[sta_id].keyinfo.conf = keyconf;
 	priv->stations[sta_id].keyinfo.keylen = keyconf->keylen;
 
 	memcpy(priv->stations[sta_id].keyinfo.key, keyconf->key,
